### PR TITLE
LibJS: Consistently call observable Temporal AOs with undefined options

### PR DESCRIPTION
**Pull request by linusg**
`filed_spec_issues++`

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -191,25 +191,33 @@ ThrowCompletionOr<Variant<String, NumberType>> get_string_or_number_option(Globa
     return value.as_string().string();
 }
 
-// 13.5 ToTemporalOverflow ( normalizedOptions ), https://tc39.es/proposal-temporal/#sec-temporal-totemporaloverflow
-ThrowCompletionOr<String> to_temporal_overflow(GlobalObject& global_object, Object const& normalized_options)
+// 13.5 ToTemporalOverflow ( options ), https://tc39.es/proposal-temporal/#sec-temporal-totemporaloverflow
+ThrowCompletionOr<String> to_temporal_overflow(GlobalObject& global_object, Object const* options)
```
Comment (ldm5180) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:195]: This changes the `const`-ness of the `Object`. Is that intentional? Why does the `options` object need to be mutable?

**Review by ldm5180 — APPROVED** [review]

File: Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp [hunk 2]
```diff
@@ -1301,46 +1271,40 @@ ThrowCompletionOr<RoundedDuration> round_duration(GlobalObject& global_object, d
         // b. Let dateAdd be ? GetMethod(calendar, "dateAdd").
         auto* date_add = TRY(Value(calendar).get_method(global_object, vm.names.dateAdd));
 
-        // c. Let firstAddOptions be OrdinaryObjectCreate(null).
-        auto* first_add_options = Object::create(global_object, nullptr);
-
-        // d. Let yearsMonthsLater be ? CalendarDateAdd(calendar, relativeTo, yearsMonths, firstAddOptions, dateAdd).
-        auto* years_months_later = TRY(calendar_date_add(global_object, *calendar, relative_to, *years_months, first_add_options, date_add));
+        // d. Let yearsMonthsLater be ? CalendarDateAdd(calendar, relativeTo, yearsMonths, undefined, dateAdd).
+        auto* years_months_later = TRY(calendar_date_add(global_object, *calendar, relative_to, *years_months, nullptr, date_add));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp:1275]: missing `c.`?

**Review by IdanHo — COMMENTED** [review]
Looks good (also nice find!), just one comment nitpick.

File: Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp:1275]: good catch!

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp:1275]: (not missing, just not properly updated after removing one)

**Pull request merged**